tests/test_markers.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+import logging
+import pathlib
+from pprint import pformat as pf
+from typing import Final
+
+import pytest
+import tomli
+from tasks import MARKER_DEPENDENDENCY_MAP
+
+pytestmarks = [pytest.mark.project]
+
+LOGGER: Final = logging.getLogger(__name__)
+PROJECT_ROOT: Final = pathlib.Path(__file__).parent.parent
+PYPROJECT_TOML: Final = PROJECT_ROOT / "pyproject.toml"
+
+
+@pytest.fixture(scope="module")
+def pyproject_toml_dict() -> dict:
+    """Parse pyporject.toml and return as dict"""
+    return tomli.loads(PYPROJECT_TOML.read_text())
+
+
+@pytest.fixture(scope="module")
+def pytest_markers(pyproject_toml_dict: dict) -> list[str]:
+    """Return pytest markers"""
+    LOGGER.debug(
+        f"pytest config ->\n{pf(pyproject_toml_dict['tool']['pytest'], depth=2)}"
+    )
+    marker_details = pyproject_toml_dict["tool"]["pytest"]["ini_options"]["markers"]
+    LOGGER.debug(f"marker_details ->\n{pf(marker_details)}")
+    return [m.split(":")[0] for m in marker_details]
+
+
+def test_marker_mappings_are_registered(pytest_markers: list[str]):
+    """
+    Check that all pytest marker mappings are acutallly valid,
+    and have been registered with pytest.
+    """
+    print(f"pytest_markers:\n----------\n{pf(pytest_markers)}")
```

Comment:
Should this be a LOGGER statement?

Reply:
thanks for catching. fixed

Reply:
I would say no, mixing logs and prints in a test functions makes it harder to separate the normal application logging from the test logging.
For this reason, the only time I tend to use a logger in tests, is in my test doubles, because it's useful to see those logs in the context of normal application logs.